### PR TITLE
Move init_env code into after_initialize hook

### PR DESCRIPTION
**Pull request by imtayadeway**
This helps to isolate things a bit better  - I'd like to be able to (at least) in theory require lib/api.rb without having to load all of Rails with it. The `after_initialize` hook seems like a better place to put this code, rather than invoking it at the time when lib/api.rb is required

/cc @jrafanie 
@miq-bot add-label technical debt
@miq-bot assign @abellotti

### CODE REVIEW

File: lib/manageiq/api/engine.rb [hunk 1]
```diff
@@ -7,6 +7,21 @@ class Engine < ::Rails::Engine
...
+        $api_log.info("Static Configuration")
+        ::Api::ApiConfig.base.each { |key, val| log_kv(key, val) }
+
+        $api_log.info("")
+        $api_log.info("Dynamic Configuration")
+        ::Api::Environment.user_token_service.api_config.each { |key, val| log_kv(key, val) }
+      end
+
+      def self.log_kv(key, val)
+        $api_log.info("  #{key.to_s.ljust([24, key.to_s.length].max, ' ')}: #{val}")
```
Comment (Fryguy) [inline lib/manageiq/api/engine.rb]: We have a thing in VMDBLogger that logs hashes, which might just simplify this...

```
$api_log.log_hashes(::Api::ApiConfig.base)
```

**Comment by Fryguy** [comment]
To avoid The GlobalVars rubocop, just update your codeclimate.local file to ignore $api_log

**Comment by imtayadeway** [comment]
@Fryguy thanks for suggestions and extra ✂️ !

**Comment by Fryguy** [comment]
LGTM :+1:... I'll leave it to @abellotti to review and merge.

**Comment by abellotti** [comment]
Thanks @imtayadeway for updating the PR.  🎹

**Pull request merged**